stl/inc/deque

```diff
@@ -1543,29 +1555,32 @@ private:
 
         size_type _Allocsize = _Newsize;
 
-        size_type _Myboff = _Myoff() / _Block_size;
-        _Mapptr _Newmap   = _Allocate_at_least_helper(_Almap, _Allocsize);
-        _Mapptr _Myptr    = _Newmap + _Myboff;
+        const auto _Myboff  = static_cast<size_type>(_Myoff() / _Block_size);
+        const auto _Map_off = static_cast<_Map_difference_type>(_Myboff);

```

Comment:
It might be better for clarity to replace `_Map_off` and `_Map_count` variables with on-site `static_cast`s here.

I have no idea about whether it's worthwhile to replace `_Map_distance` with on-site `static_cast`s, though.